pkg/cloudprovider/aws/apis/v1alpha1/provider_defaults.go

```diff
@@ -65,3 +69,36 @@ func (c *Constraints) defaultAMIFamily() {
 	}
 	c.AMIFamily = &AMIFamilyAL2
 }
+
+func (c *Constraints) defaultBlockDeviceMappings() {
+	if c.LaunchTemplate != nil {
+		return
+	}
+	if len(c.BlockDeviceMappings) != 0 {
```

Comment:
I like this simplification w/ no merging. If you have an opinion, you have to get it right.